cylc/flow/scheduler_cli.py

```diff
@@ -390,11 +390,7 @@ async def scheduler_cli(
 
     # check the workflow can be safely restarted with this version of Cylc
     db_file = Path(get_workflow_srv_dir(workflow_id), 'db')
-    if not _version_check(
-        db_file,
-        options.upgrade,
-        options.downgrade,
-    ):
+    if not _version_check(db_file, options):
```

Comment:
FYI, I purposefully did *not* propagate the `options` object down the call chain. IMO it's a good idea to abstract out the CLI interfaces higher up in the call chain to keep the underlying code more abstract. The more you propagate CLI logic down the call stack, the harder it is to, for example, open a Python API later, or create a wrapper command, you also need to mock more in order to test and typing is less effective too (we know we're getting a `Values` object, but we don't know whether it has an `upgrade` option.

Not a blocker, but not positive "tidying" IMO.

Reply:
I think ideally we want to only alter `options.upgrade` in the case of an interactive upgrade, so the simplest way of doing this is passing in the `options` object to `_version_check`. 

It is a shame about the type-looseness of it, but at least it is relatively easy to test these functions by importing and using `RunOptions` (no need for mocking).

https://github.com/cylc/cylc-flow/blob/6eb8f61c5b9777cb21ac68eff13aa4f0722f691a/cylc/flow/option_parsers.py#L569-L570

Reply:
`options.update = _version_check(db_file, options.upgrade, options.downgrade)` would have done it?

Reply:
There are pathways it returns `True` without interactive upgrade